ecl/hql/hqlfold.cpp

```diff
@@ -596,8 +596,13 @@ bool checkExternFoldable(IHqlExpression* expr, unsigned foldOptions, StringBuffe
     unsigned numParam = expr->numChildren();
     for(unsigned iparam = 0; iparam < numParam; iparam++)
     {
-        if (!expr->queryChild(iparam)->queryValue())            //NB: Already folded...
+        switch (expr->queryChild(iparam)->getOperator())
+        {
+        case no_constant: case no_null: case no_all:
```

Comment:
minor: no_all isn't supported now, but should be once you extend what can be folded it will be.